level1/task4.php

```diff
@@ -1,72 +1,131 @@
 <?php
+require_once 'base.php';
+require_once 'responses.php';
 
-require_once "base.php";
-require_once "responses.php";
-
+/**
+ * The success return value for authentication.
+ */
 const SUCCESS_RETURN_VALUE = '<h1 style="color:green">FOUND</h1>';
-const USERS_CREDENTIALS = "./files/passwords.txt";
-const CREDENTIALS_SEPARATOR = ":";
-const AUTH_URL = "/api/checkLoginAndPassword";
-const REQUIRED_CONTENT_TYPE = "application/x-www-form-urlencoded";
 
-function getContentType($headers)
-{
-    $contentType = "";
+/**
+ * File containing user credentials.
+ */
+const USERS_CREDENTIALS = './files/passwords.txt';
+
+/**
+ * Separator used in the credentials file to separate login and password.
+ */
+const CREDENTIALS_SEPARATOR = ':';
+
+/**
+ * URL for the authentication endpoint.
+ */
+const AUTH_URL = '/api/checkLoginAndPassword';
 
+/**
+ * Required content type for the request body.
+ */
+const REQUIRED_CONTENT_TYPE = 'application/x-www-form-urlencoded';
+
+/**
+ * Retrieves the Content-Type header from an array of headers.
+ *
+ * @param array $headers The array of headers.
+ * @return string|null The Content-Type header value.
+ */
+function getContentType(array $headers): ?string
+{
     foreach ($headers as $header) {
-        if ($header[0] == "Content-Type") {
-            $contentType = $header[1];
+        if ($header[0] == 'Content-Type') {
+            return $header[1];
         }
     }
 
-    return $contentType;
+    return null;
+}
+
+/**
+ * Extracts the login and password from the request body.
+ *
+ * @param string $body The request body.
+ * @return array An array containing login and password.
+ */
+function getAuthCredentials(string $body): array
+{
+    $parts = explode('&', $body);
+
+    return [
+        explode('=', $parts[0])[1],
```

Comment:
do it like `['login' => ..., 'password' => ...]`
code  will be more understandable